pandas/tests/extension/test_arrow.py

```diff
@@ -2150,14 +2150,39 @@ def test_str_rsplit():
     tm.assert_frame_equal(result, expected)
 
 
-def test_str_unsupported_extract():
-    ser = pd.Series(["abc", None], dtype=ArrowDtype(pa.string()))
-    with pytest.raises(
-        NotImplementedError, match="str.extract not supported with pd.ArrowDtype"
-    ):
+def test_str_extract_non_symbolic():
+    ser = pd.Series(["a1", "b2", "c3"], dtype=ArrowDtype(pa.string()))
+    with pytest.raises(ValueError, match="pat=.* must contain a symbolic group name."):
         ser.str.extract(r"[ab](\d)")
 
 
+def test_str_extract():
+    ser = pd.Series(["a1", "b2", "c3"], dtype=ArrowDtype(pa.string()))
+    result = ser.str.extract(r"(?P<letter>[ab])(?P<digit>\d)")
+    expected = pd.DataFrame(
+        {
+            "letter": ArrowExtensionArray(pa.array(["a", "b", None])),
+            "digit": ArrowExtensionArray(pa.array(["1", "2", None])),
+        }
+    )
+    tm.assert_frame_equal(result, expected)
+
+
+def test_str_extract_expand():
+    ser = pd.Series(["a1", "b2", "c3"], dtype=ArrowDtype(pa.string()))
+    result = ser.str.extract(r"[ab](?P<digit>\d)", expand=True)
+    expected = pd.DataFrame(
+        {
+            "digit": ArrowExtensionArray(pa.array(["1", "2", None])),
+        }
+    )
+    tm.assert_frame_equal(result, expected)
+
+    result = ser.str.extract(r"[ab](?P<digit>\d)", expand=False)
```

Comment:
Can we add coverage for when there are multiple groups and `expand=False`?

Reply:
Sure, I parametrized the above test with `expand=`

Reply:
Shouldn't the `expand=False` case only produce a series?

Reply:
Not necessarily:

https://pandas.pydata.org/docs/reference/api/pandas.Series.str.extract.html

> If True, return DataFrame with one column per capture group. If False, return a Series/Index if there is one capture group or DataFrame if there are multiple capture groups.

Reply:
Ah OK - sorry I misremembered how that works. I was somewhat expecting `expand=False` here to just return the struct array

Reply:
FWIW I'm OK with this now too as it is definitely consistent with our current API. There may be a future where we actuallly want it to return a pyarrow struct but can always come back and do that later

Reply:
Yeah agreed. For arrow types, I think in the future `expand=False` should return a struct typed Series